ndk/src/font.rs

```diff
@@ -0,0 +1,401 @@
+        write!(
+            f,
+            "{}{}{}{}",
+            bytes[0] as char, bytes[1] as char, bytes[2] as char, bytes[3] as char
+        )
+    }
+}
+
+impl fmt::Debug for AxisTag {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "AxisTag(")?;
+        fmt::Display::fmt(self, f)?;
+        write!(f, " {:#x})", self.0)
+    }
+}
+
+/// The error type returned when an invalie font weight value is passed.
+#[derive(Debug)]
+pub struct TryFromU32Error(());
+
+impl fmt::Display for TryFromU32Error {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            fmt,
+            "each byte in an axis tag must be in the range 0x20 to 0x7E"
+        )
+    }
+}
+
+impl std::error::Error for TryFromU32Error {}
+
+/// A native [`AFont *`]
+///
+/// [`AFont *`]: https://developer.android.com/ndk/reference/group/font#afont_close
+#[cfg(feature = "api-level-29")]
+#[derive(Debug)]
+pub struct Font {
+    ptr: NonNull<ffi::AFont>,
+}
+
+#[cfg(feature = "api-level-29")]
+impl Font {
+    /// Create an `Font` from a pointer
+    ///
+    /// # Safety
+    /// By calling this function, you assert that the pointer is a valid pointer to a native
+    /// `AFont`.
+    pub unsafe fn from_ptr(ptr: NonNull<ffi::AFont>) -> Self {
+        Self { ptr }
+    }
+
+    /// Returns the pointer to the native `AFont`.
+    pub fn ptr(&self) -> NonNull<ffi::AFont> {
+        self.ptr
+    }
+
+    /// Return a count of font variation settings associated with the current font.
+    ///
+    /// The font variation settings are provided as multiple tag-values pairs.
+    ///
+    /// For example, bold italic font may have following font variation settings: 'wght' 700,
+    /// 'slnt' -12. In this case, [`Font::axis_count`] returns 2 and [`Font::axis_tag_at`] and
+    /// [`Font::axis_value_at`] will return following values.
+    pub fn axis_count(&self) -> usize {
+        unsafe { ffi::AFont_getAxisCount(self.ptr.as_ptr()) }
+    }
+
+    /// Return an OpenType axis tag associated with the current font.
+    ///
+    /// See [`Font::axis_count`] for more details.
+    pub fn axis_tag_at(&self, idx: usize) -> AxisTag {
+        // Android returns Axis Tag in big-endian.
+        // See https://cs.android.com/android/platform/superproject/+/refs/heads/master:frameworks/base/native/android/system_fonts.cpp;l=197 for details
+        AxisTag(unsafe { ffi::AFont_getAxisTag(self.ptr.as_ptr(), idx as u32) })
+    }
+
+    /// Return an OpenType axis value associated with the current font.
+    ///
+    /// See [`Font::axis_count`] for more details.
+    pub fn axis_value_at(&self, idx: usize) -> f32 {
+        unsafe { ffi::AFont_getAxisValue(self.ptr.as_ptr(), idx as u32) }
+    }
+
+    /// Return a font collection index value associated with the current font.
+    ///
+    /// In case the target font file is a font collection (e.g. .ttc or .otc), this returns a non
```

Comment:
```suggestion
    /// In case the target font file is a font collection (e.g. `.ttc` or `.otc`), this returns a non
```
And isn't it `non-negative` with a hyphen?

Reply:
It is. Fixed!